### PR TITLE
Fix syntax for extra containers for ensemblvep and snpeff

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/ensemblvep/main.nf [hunk 1]
```diff
@@ -2,13 +2,11 @@ process ENSEMBLVEP {
     label 'process_medium'
 
     conda (params.enable_conda ? "bioconda::ensembl-vep=104.3" : null)
-    if (task.ext.use_cache) {
-        container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        !task.ext.use_cache ?
```
Comment (maxulysse) [inline modules/ensemblvep/main.nf:6]:
```suggestion
        task.ext.use_cache ?
```

File: modules/snpeff/main.nf [hunk 2]
```diff
@@ -2,13 +2,11 @@ process SNPEFF {
     label 'process_medium'
 
     conda (params.enable_conda ? "bioconda::snpeff=5.0" : null)
-    if (task.ext.use_cache) {
-        container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        !task.ext.use_cache ?
```
Comment (maxulysse) [inline modules/snpeff/main.nf:6]:
```suggestion
        task.ext.use_cache ?
```

File: modules/ensemblvep/main.nf [hunk 3]
```diff
@@ -25,7 +23,7 @@ process ENSEMBLVEP {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
```
Comment (FriederikeHanssen) [inline modules/ensemblvep/main.nf]: This needs to be updated with the new suffix annotation. I think like this:
```suggestion
    def suffix = task.ext.prefix ?: "${task.ext.prefix}"
```

File: modules/snpeff/main.nf [hunk 4]
```diff
@@ -29,14 +27,14 @@ process SNPEFF {
         avail_mem = task.memory.giga
     }
     def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
```
Comment (FriederikeHanssen) [inline modules/snpeff/main.nf]: same here: 
```suggestion
    def suffix = task.ext.prefix ?: "${task.ext.prefix}"
```

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**